From 18dcaa70d4cd2cff75f3eda237b7ac7eea8398d3 Mon Sep 17 00:00:00 2001
From: Vladimir Barinov <vladimir.barinov@cogentembedded.com>
Date: Wed, 6 Oct 2021 12:29:22 +0300
Subject: [PATCH] media: i2c: ar0233: fix multicamera usecase for 30fps

This fixes the multicamera use case. it requires larger
blanking

Signed-off-by: Vladimir Barinov <vladimir.barinov@cogentembedded.com>
---
 drivers/media/i2c/soc_camera/ar0233.c      | 2 ++
 drivers/media/i2c/soc_camera/ar0233_rev2.h | 4 ++--
 2 files changed, 4 insertions(+), 2 deletions(-)

diff --git a/drivers/media/i2c/soc_camera/ar0233.c b/drivers/media/i2c/soc_camera/ar0233.c
index c6ede00..9286da3 100644
--- a/drivers/media/i2c/soc_camera/ar0233.c
+++ b/drivers/media/i2c/soc_camera/ar0233.c
@@ -642,6 +642,8 @@ static int ar0233_initialize(struct i2c_client *client)
 						ar0233_regs_seplus_mipi_12bit_60fps_rev2[9] = ar0233_rev2_empty;
 					}
 
+					ar0233_regs_seplus_mipi_12bit_60fps_rev2[2] = ar0233_rev2_pll_27_102_4lane_12b;
+					ar0233_regs_seplus_mipi_12bit_60fps_rev2[3] = ar0233_rev2_mipi_12bit_4lane;
 					ar0233_regs_seplus_mipi_12bit_60fps_rev2[10] = ar0233_rev2_SE_Lin_T2_12bit_30fps;
 					ar0233_set_regs(client, ar0233_regs_seplus_mipi_12bit_60fps_rev2);
 					if (extclk != 27) {
diff --git a/drivers/media/i2c/soc_camera/ar0233_rev2.h b/drivers/media/i2c/soc_camera/ar0233_rev2.h
index 0ac82d2..94dc6c8 100644
--- a/drivers/media/i2c/soc_camera/ar0233_rev2.h
+++ b/drivers/media/i2c/soc_camera/ar0233_rev2.h
@@ -3483,8 +3483,8 @@ static const struct ar0xxx_reg ar0233_rev2_SE_Lin_T2_12bit_30fps[] = {
 {0x30BA, 0x1121}, //Num_exp_max
 {0x31AC, 0x140C}, //12 bit output
 
-{0x300C, 0x0760}, // LINE_LENGTH_PCK_
-{0x300A, 0x089e}, // FRAME_LENGTH_LINES_
+{0x300A, AR0233_SENSOR_HEIGHT + 100}, // Frame_length_Lines
+{0x300C, AR0233_SENSOR_WIDTH + 400}, // Line_length_pck
 {0x33E0, 0x0D40}, // TEST_ASIL_ROWS_idle row=7
 {0x3362, 0x0001}, // DC_GAIN
 {0x3238, 0x8446}, // EXPOSURE_RATIO
-- 
2.7.4

